### PR TITLE
Improv/acceptance criteria fix

### PR DESCRIPTION
**Pull request by adiletbaimyrza**
introduced features to pass new acceptance criteria

### CODE REVIEW

**Review by KHarpushkina — CHANGES_REQUESTED** [review]
1) In CoreIcon.spec.js. You can use setProps to change props for the component. You don't need to call shallowMount for this purpose
2) In NewTodo.spec.js. In the test 'fires addNewTodo method on keyup.enter', I think it's better to check if the 'add' mutation was called or not. You can write a couple of tests checking when it should be called and when it shouldn't. You don't need to mock the component methods.
3) The same in HeaderBar.spec.js, in the test 'fires toggleTheme method when button is clicked'. Try to create spyOn on useToggle and check if it was called.
4) Also not sure if checks expect(wrapper.exists()).toBe(true) are needed, I guess snapshots already do that job.
5) In useToggle.spec.ts you didn't add a test for the case when new error should be thrown (
if (typeof targetRef !== 'object' || !('value' in targetRef)) {
    throw new TypeError('targetRef must be a ref object')
  }
)

Unit tests looks good!

File: src/components/CoreIcon/CoreIcon.spec.js [hunk 1]
```diff
@@ -1,38 +1,39 @@
...
-  const propsWithoutSizes = {
-    src: 'test-icon.png',
-  }
-  it('renders correctly', () => {
-    const wrapper = shallowMount(CoreIcon, {
-      props: propsWithoutSizes,
+  let wrapper
+
+  beforeEach(() => {
+    wrapper = mount(CoreIcon, {
```
Comment (KHarpushkina) [inline src/components/CoreIcon/CoreIcon.spec.js:9]: Why did you decide to use mount? In most cases, it is recommended to use shallowMount for unit tests, as it simplifies the tests, reduces the impact of external factors (no dependencies on child components) and makes them run faster.

**Review by KHarpushkina — CHANGES_REQUESTED** [review]
You haven't applied Atomic Design to your project structure. Do you plan to do it in future PR?

File: src/components/CoreIcon/CoreIcon.spec.js (hunk 1 above)
Comment (adiletbaimyrza) [reply src/components/CoreIcon/CoreIcon.spec.js:9]: >Like [mount](https://v1.test-utils.vuejs.org/api/#mount), it creates a [Wrapper](https://v1.test-utils.vuejs.org/api/wrapper) that contains the mounted and rendered Vue component, but with stubbed child components.

>Notice that using shallowMount will make the component under testing different from the component you run in your application – some of its parts won't be rendered! This is why it is not the suggested way of testing components unless you face performance issues or need to simplify test arrangements.



This is from documentation of vue test utils, which doesn't suggest using shallowMount

basically I realized the component doesn't have children and no need for stubbing child components, so I changed it to mount

Comment (adiletbaimyrza) [reply src/components/CoreIcon/CoreIcon.spec.js:9]: I understand the concept atomic design, I am just wondering why we are following that strategy in developing ui components paritcularly for this project.

**Review by KHarpushkina — APPROVED** [review]

**Pull request merged**